kano_i18n/init.py

```diff
@@ -9,8 +9,50 @@
 import gettext
 import __builtin__
 
+REGISTERED_DOMAINS = set()
+CURRENT_TRANSLATION = None
+
+
 def install(app, locale_dir=None):
+    global CURRENT_TRANSLATION
+
     # N_() defined globally for deferred translation
     __builtin__.__dict__['N_'] = lambda msg: msg
 
-    gettext.install(app, localedir=locale_dir, unicode=1)
+    trans = gettext.translation(app, localedir=locale_dir, fallback=True, codeset=None)
+    trans.install(True, names=None)
+
+    CURRENT_TRANSLATION = trans
+
+
+def register_domain(domain, locale_dir=None):
+    """
+    Register a domain as a fallback to the currently installed translation.
+    Used by library modules to register their translations on top of the current
+    app translation.
+    """
+    global REGISTERED_DOMAINS
+    global CURRENT_TRANSLATION
+
+    if not hasattr(__builtin__, '_') or not CURRENT_TRANSLATION:
+        # TODO
```

Comment:
Unsure what the best thing to do here would be. Any suggestions @Ealdwulf @tombettany ?